sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -363,8 +363,7 @@ final class TypeSig private(
       case FloatType => check.contains(TypeEnum.FLOAT)
       case DoubleType => check.contains(TypeEnum.DOUBLE)
       case DateType => check.contains(TypeEnum.DATE)
-      case TimestampType if check.contains(TypeEnum.TIMESTAMP) =>
-          TypeChecks.areTimestampsSupported()
```

Comment:
Originally invoked by shuffle meta, FileFormatChecks, tag AST and other.
- shuffle meta
- FileFormatChecks: Spark always write Parqeut with UTC timestamp, it's safe;  For ORC, Spark map ORC type `timestamp with local time zone` to Spark type TIMESTAMP_NTZ (with no time zone). Now spark-rapids does not support TIMESTAMP_NTZ currently, so it's safe to remove the check.
- tag AST: Not sure if remove this UTC check is OK, need to investigate.